src/depackers/stb_vorbis.c

```diff
@@ -741,12 +1121,14 @@ static int compute_codewords(Codebook *c, uint8 *len, int n, uint32 *values)
       }
 
       while (z > 0 && !available[z]) --z;
-      if (z == 0) { /*assert(0);*/ return FALSE; }
+      if (z == 0) { return FALSE; }
       res = available[z];
+      assert(z >= 0 && z < 32);
       available[z] = 0;
       add_entry(c, bit_reverse(res), i, m++, len[i], values);
-      // propogate availability up the tree
+      // propagate availability up the tree
       if (z != len[i]) {
+         assert(len[i] < 32);
          for (y=len[i]; y > z; --y) {
             assert(available[y] == 0);
```

Comment:
why not return failure somehow and assert() instead?
(yeah, this was here before the new version..)

Reply:
It does seem like this line of code should be an actual check and not an assert. I'll dig a little bit and see if there's a reason why it isn't...

Reply:
OK, the answer is at line 3834: the lengths values are initialized with `get_bits(f, 5) + 1` and a value of 32 is considered an error case. So I think this is technically fine.